src/components/utils/useElementSize.ts

```diff
@@ -0,0 +1,54 @@
+import {useLayoutEffect, useState, MutableRefObject} from 'react';
+import ResizeObserver from 'resize-observer-polyfill';
+import _throttle from 'lodash/throttle';
+
+const RESIZE_THROTTLE = 16;
+
+interface ElementSize {
+    width: number;
+    height: number;
+}
+
+export function useElementSize<T extends HTMLElement = HTMLDivElement>(
+    ref: MutableRefObject<T | null> | null,
+) {
+    const [size, setSize] = useState<ElementSize>({
+        width: 0,
+        height: 0,
+    });
+
+    useLayoutEffect(() => {
+        if (!ref?.current) {
+            return undefined;
+        }
+
+        const handleResize: ResizeObserverCallback = (entries) => {
+            if (!Array.isArray(entries)) {
+                return;
+            }
+
+            const entry = entries[0];
+            if (entry.borderBoxSize?.length > 0) {
```

Comment:
from mdn [example](https://github.com/mdn/dom-examples/blob/main/resize-observer/resize-observer-text.html#L88), there might be an object here.